src/material-experimental/mdc-input/harness/input-harness.ts

```diff
@@ -0,0 +1,116 @@
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: InputHarnessFilters = {}): HarnessPredicate<MatInputHarness> {
+    return new HarnessPredicate(MatInputHarness)
+        .addOption(
+            'name', options.name, async (harness, name) => (await harness.getName()) === name)
+        .addOption('id', options.id, async (harness, id) => (await harness.getId()) === id)
+        .addOption(
+            'value', options.value, async (harness, value) => (await harness.getValue()) === value);
+  }
+
+  /** Whether the input is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return (await this.host()).getPropertyValue('disabled')!;
+  }
+
+  /** Whether the input is required. */
+  async isRequired(): Promise<boolean> {
+    return (await this.host()).getPropertyValue('required')!;
+  }
+
+  /** Whether the input is readonly. */
+  async isReadonly(): Promise<boolean> {
+    return (await this.host()).getPropertyValue('readOnly')!;
+  }
+
+  /** Gets the value of the input. */
+  async getValue(): Promise<string> {
+    // The "value" property of the native input is never undefined.
+    return (await (await this.host()).getPropertyValue('value'))!;
+  }
+
+  /** Gets the name of the input. */
+  async getName(): Promise<string> {
+    // The "name" property of the native input is never undefined.
+    return (await (await this.host()).getPropertyValue('name'))!;
+  }
+
+  /**
+   * Gets the type of the input. Returns "textarea" if the input is
+   * a textarea.
+   */
+  async getType(): Promise<string> {
+    // The "type" property of the native input is never undefined.
+    return (await (await this.host()).getPropertyValue('type'))!;
+  }
+
+  /** Gets the placeholder of the input. / */
+  async getPlaceholder(): Promise<string> {
+    // The "placeholder" property of the native input is never undefined.
+    return (await (await this.host()).getPropertyValue('placeholder'))!;
+  }
+
+  /** Gets the id of the input. */
+  async getId(): Promise<string> {
+    // The input directive always assigns a unique id to the input in
+    // case no id has been explicitly specified.
+    return (await (await this.host()).getPropertyValue('id'))!;
+  }
+
+  /**
+   * Focuses the input and returns a promise that indicates when the
+   * action is complete.
+   */
+  async focus(): Promise<void> {
+    return (await this.host()).focus();
+  }
+
+  /**
+   * Blurs the input and returns a promise that indicates when the
+   * action is complete.
+   */
+  async blur(): Promise<void> {
+    return (await this.host()).blur();
+  }
+
+  /**
+   * Sets the value of the input. The value will be set by simulating
+   * keypresses that correspond to the given value.
+   */
+  async setValue(newValue: string): Promise<void> {
+    const inputEl = await this.host();
+    await inputEl.clear();
+    // We don't want to send keys for the value if the value is an empty
```

Comment:
should we update `sendKeys` to not do that?

Reply:
I'm not sure. I could see `sendKeys` being a noop in that case. Though it also feels like an anti-pattern in general to call `sendKeys` with an empty value.

Reply:
Yeah, calling it with empty string is weird and people shouldn't do it, so I'm fine with changing it or just leaving it how it is

Reply:
I'd vote for leaving it as is, but maybe throwing an error in `sendKeys` instead? (follow-up)?